sensor/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,140 @@
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization2.jpg"  alt="visualization2">
+</p>
+
+Once the user gets the converged detection, the user can start moving the tag to another position. Please make sure the moving distance is larger than the `calibration_min_pair_distance` and also make sure the tag is in the FOV of the lidar and camera.
+
+At the end of the calibration, we can get 8 detection pairs which are shown below.
+
+![segment](../images/tag_based_pnp_calibrator/visualization3.jpg)
+
+## Results
+
+After the calibration process is finished, the `sensor_calibration_manager` will display the results in the tf tree and allow the user to save the calibration data to a file.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu4.jpg" alt="menu4" width="500">
+</p>
+
+The user can modify the `visualization options` on the right side of the `image view`. To compare the results, please set the `Marker size (m)` to `0.04` and set the `PC subsample factor` to `1`.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization_bar.jpg"  alt="visualization_bar" width="200">
+</p>
+
+After setting the options above, change the `/initial_tf` (in the `visualization options`) to `/current_tf`. By doing this, it is easier to measure the difference after the calibration.
+
+The images below show that with the calibrated transformation, the projected pointcloud aligns better with the image.
+
+<table>
+  <tr>
+    <td><img src="../images/tag_based_pnp_calibrator/init_tf.jpg" alt="init_tf" width = 500px ></td>
+    <td><img src="../images/tag_based_pnp_calibrator/calibrated_tf.jpg" alt="calibrated_tf" width = 500px ></td>
+   </tr>
+   <tr>
+    <td><p style="text-align: center;">Before Calibration.</p></td>
+    <td><p style="text-align: center;">After Calibration.</p></td>
+  </tr>
+</table>
+
+## FAQ
+
+- Why the calibrator doesn't add calibration pairs?
+
+  1. One possible reason is that the current pair is too close to previously collected data. In that case, the current data is not accepted.
+  2. The timestamps of the lidar and camera are not synchronized, this can be checked with `ros2 topic echo [topic_name]`. Setting the parameter `use_receive_time` to `True` might help to solve the issue.
+  3. The detections are not stable enough (detections don’t converge).
+
+- Why the UI doesn't launch?
+
+  1. Check with `ros2 node list` if the relevant nodes have been started. It is possible that the provided parameters don’t match any of the valid arguments
+  2. If the UI crashes (see the console), it is probably due to bad PySide installation, invalid intrinsic parameters, or invalid extrinsic parameters.
+  3. The timestamps of the lidar and camera are not synchronized.
+
+- Why the reprojection errors are so high
```

Comment:
Other parts of this sections are presented as questions. Please make it consistent.

If detections look Ok and we have several of them, sure, intrinsics.
If we have outliers it also matches the high reprojection error

Reply:
Thanks! fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/32dd61431d3beb3d84ecb085cd246042de8efb09